### DISCUSSION TITLE
Which device will work, and how to configure it?

### DISCUSSION THREAD

**Discussion by JohnLussmyer** [collin80/SavvyCAN#769] [General]
I have 3 USB interfaces, and I'm trying to figure out how to use one with SaavyCan.  I can use either a Win 10 PC, or Linux on a Raspberry Pi.
1) Ginkgo/Viewtool USB CAN Adapter 
2) Microchip Can Bus Analyzer (USB) 
3) Dual Can Hat on a Raspberry Pi 4. (Using Javacan with a custom app right now)

The main thing I need to do is to be able to capture a can log in .asc format.
(Heck, if somebody could provide a good description of the .asc can log format, I could write it directly.  Haven't found that info anywhere though.)

**Comment by collin80** [comment]
The easiest approach may be linux as most any CAN device will be a socketcan device on linux so then you can more easily interface CAN hardware to SavvyCAN. On Windows you need specific support for each device and it is lacking generally speaking. SavvyCAN can read Microchip CANBus Analyzer log files but it won't connect to the actual device itself.

SavvyCAN will write .ASC files so you can use it for that. In fact, since it reads Microchip log files you could even perhaps log in the Microchip program, read the log in SavvyCAN, and output it to Vector ASC format. Since SavvyCAN can write the format you can also read the source code to get an idea of how to write ASC files yourself. The ASC saving code is here: https://github.com/collin80/SavvyCAN/blob/0447ec1435e98420f9afaca0a5e55728e02278e0/framefileio.cpp#L1881

But, SavvyCAN will run perfectly fine on a Raspberry Pi. Lots of people have done it. I just don't have a binary you can use, you'd have to compile it yourself. Still, you can compile it for RPi4 and run it directly on the Pi with the CAN hat and then save files that way too.
